### ISSUE TITLE
init_numbered_fields incorrectly triggers for type alias of tuple struct

### ISSUE THREAD

**Issue by nyospe** [rust-lang/rust-clippy#8638]
### Summary

The clippy lint `init_numbered_fields` introduced in 1.59.0 is emitted with an incorrect suggested fix when the struct initialization is for a type alias to a tuple or unit struct.

This case should be omitted from the lint entirely, as there is no means other than use of numbered fields to initialize a tuple struct for which only the type alias can be reliably known, because the type alias cannot be used as a constructor.

### Reproducer

I tried this code:

```rust
fn main() {
    let _ = TypeAlias { 0: 0 };
}


struct AkwardlyNamedAndPossiblyCfgGatedTupleStruct(i32);

type TypeAlias = AkwardlyNamedAndPossiblyCfgGatedTupleStruct;
```

I ran `cargo clippy`

I saw this happen:

```
warning: used a field initializer for a tuple struct
 --> src/main.rs:2:13
  |
2 |     let _ = TypeAlias { 0: 0 };
  |             ^^^^^^^^^^^^^^^^^^ help: try this instead: `TypeAlias(0)`
  |
  = note: `#[warn(clippy::init_numbered_fields)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#init_numbered_fields

warning: `test_clippy_error` (bin "test_clippy_error") generated 1 warning
```

I expected to see this happen:
```
<no clippy warning>
```
because "A type alias to a tuple-struct or unit-struct cannot be used to qualify that type's constructor:", this should not trigger when the tuple struct initialization is being invoked through a type alias.

### Version

```text
rustc 1.59.0 (9d1b2106e 2022-02-23)
binary: rustc
commit-hash: 9d1b2106e23b1abd32fce1f17267604a5102f57a
commit-date: 2022-02-23
host: x86_64-apple-darwin
release: 1.59.0
LLVM version: 13.0.0
```

**Comment by nyospe** [comment]
Confirmed still present in 1.60.0

**Comment by nyospe** [comment]
Note: introduced by https://github.com/rust-lang/rust-clippy/pull/8170
